### PR TITLE
Add repeat and tile ops

### PR DESCRIPTION
**Pull request by reminisce**

### CODE REVIEW

**Comment by piiswrong** [comment]
We need backward for these two op. Could you
1. refactor repeat to use similar mechanism as tile
2. add backward by calling ReduceAxesCompute

**Comment by sxjscience** [comment]
I find that repeat and tile could be implemented by combining broadcast_to and reshape. Could we call the current implementations instead of writing a new kernel?

**Comment by reminisce** [comment]
@sxjscience Thanks for the suggestion. That's @piiswrong 's first comment as well. I will submit the commits soon.

File: tests/python/unittest/test_operator.py [hunk 1]
```diff
@@ -2477,6 +2477,147 @@ def test_cast():
...
+                    k = j * repeats
+                    expected_grad[i][j] = sum(npout_grad[i, k:k + repeats])
+        else:
+            raise RuntimeError("Invalid axis value")
+
+        assert_almost_equal(expected_grad, arr_grad.asnumpy(), threshold=5e-4)
+
+    test_repeat_forward()
+    test_repeat_backward(axis=0)
+    test_repeat_backward(axis=1)
```
Comment (piiswrong) [inline tests/python/unittest/test_operator.py]: please add a test using check numerical gradient

**Pull request merged**